packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/Convert.java

```diff
@@ -494,6 +513,14 @@ static Point pointFromPigeon(Messages.PlatformPoint point) {
     return new Point(point.getX().intValue(), point.getY().intValue());
   }
 
+  @Nullable
+  static Point pointFromPigeon(@Nullable Messages.PlatformOffset point, float density) {
+    if (point == null) {
+      return null;
+    }
+    return new Point((int) (point.getDx() * density), (int) (point.getDy() * density));
```

Comment:
why should this be a truncation (cast to int) instead of a round?

Reply:
The previous implementation replaced by this PR called `toPoint`, which called `toPixels`, which performed a cast to int. I did not intend this PR to change any implementation details beyond how the data is structured.

Reply:
I believe that this is a logic change. toPixels calls toFractionalPixels which multiplies a float by the density. 
```
  private static float toFractionalPixels(Object o, float density) {
    return toFloat(o) * density;
  }
```

In none of the code I traced did I see us dropping data by an int cast.

Reply:
`toPixels` converts the result of `toFractionalPixels` to int via (int) cast: https://github.com/flutter/packages/pull/7507/files#diff-26752280ec885e4465baf81a4771b5a7e6fd1041d333c9fedd712bcc5c3916c7L534.  
`toPoint` calls `toPixels`: https://github.com/flutter/packages/pull/7507/files#diff-26752280ec885e4465baf81a4771b5a7e6fd1041d333c9fedd712bcc5c3916c7L568.  
The `zoomBy` case previously called `toPoint`: https://github.com/flutter/packages/pull/7507/files#diff-26752280ec885e4465baf81a4771b5a7e6fd1041d333c9fedd712bcc5c3916c7L363.  
The new `zoomBy` case calls `pointFromPigeon`: https://github.com/flutter/packages/pull/7507/files#diff-26752280ec885e4465baf81a4771b5a7e6fd1041d333c9fedd712bcc5c3916c7L363.

Reply:
I am mistaken, resolving.